docs/compatibility_draft.md

```diff
@@ -0,0 +1,246 @@
+# StableHLO Compatibility
+
+## Terminology
+
+**Backward compatibility:** Programs serialized by an old version of libStablehlo can be deserialized by a new version of libStablehlo.
+
+*Example:* User serializes their StableHLO program. Over the next few months, libStablehlo receives several op updates. Regardless of these updates, the old serialized program can still be deserialized with the new version of libStablehlo.
+
+**Forward compatibility:** Programs serialized by a new version of libStablehlo can be deserialized by an old version of libStablehlo, unless they are using new features introduced since the old version.
+
+*Example:* User deploys binary that links against libStablehlo. A week later, a new op is added to StableHLO. User updates their libStablehlo locally, but does not update the binary. Any StableHLO programs serialized locally using the new version of libStablehlo that do not use this new op, can be deserialized by the old version of libStablehlo in the deployed binary.
+
+## Requirements
+We've interviewed various groups using MHLO, and based upon their use cases have collected the following requirements for StableHLO. Please let us know if you have additional requirements or would like to update existing requirements:
+
+|  R#   | Description | Priority |
+|:-----:|-------------|:--------:|
+|   1   | Must provide a minimum of 6 months backwards compatibility for serialized StableHLO programs. | Must have |
+|   2   | Must provide a minimum of 3 weeks forwards compatibility for serialized StableHLO programs. | Must have |
+|   3   | C++ and Python APIs should provide some degree of source compatibility. | Nice to have |
+
+## Compatibility Guarantees
+**Phase 1: Static Guarantees**
+
+In the short term StableHLO will provide static guarantees on stability. This includes IR structure, verifiers, and type inference. This will be expanded to include runtime tests as the interpreter becomes available. 
+
+*Backward compatibility:* After upgrade, the v1 file must match the semantics of the v2 file.
+
+*Forward compatibility:* After downgrade, the v2 file must match the semantics of the v1 file.
+
+**Phase 2: Dynamic Guarantees**
+
+A long term goal of this project is to verify result parity using the reference StableHLO interpreter.
+
+*Backward compatibility:* After upgrade, the v1 file must produce the same results or errors as the v2 file.
+
+*Forward compatibility:* After downgrade, the v2 file must produce the same results or errors as the v1 file.
+
+## Supported Dialects
+
+### Operations
+
+The following list of ops are use-case driven and can grow as needed. Supported operations will be validated using a pass before bytecoding. The list of ops and dialects to initially support is as follows:
```

Comment:
Removed the "(see a discussion about the proposed ...)" section since it would only make sense during review, not after merge. Probably better to add a comment on the actual PR.